packages/flutter/test/widgets/scrollable_selection_test.dart

```diff
@@ -0,0 +1,641 @@
+    await gesture.up();
+    expect(paragraph50.selections[0], const TextSelection(baseOffset: 2, extentOffset: 4));
+
+    controller.jumpTo(0);
+    await tester.pumpAndSettle();
+    expect(find.text('Item 50'), findsNothing);
+
+    controller.jumpTo(2000);
+    await tester.pumpAndSettle();
+    expect(find.text('Item 50'), findsOneWidget);
+    paragraph50 = tester.renderObject<RenderParagraph>(find.descendant(of: find.text('Item 50'), matching: find.byType(RichText)));
+    expect(paragraph50.selections[0], const TextSelection(baseOffset: 2, extentOffset: 4));
+
+    controller.jumpTo(4000);
+    await tester.pumpAndSettle();
+    expect(find.text('Item 50'), findsNothing);
+
+    controller.jumpTo(2000);
+    await tester.pumpAndSettle();
+    expect(find.text('Item 50'), findsOneWidget);
+    paragraph50 = tester.renderObject<RenderParagraph>(find.descendant(of: find.text('Item 50'), matching: find.byType(RichText)));
+    expect(paragraph50.selections[0], const TextSelection(baseOffset: 2, extentOffset: 4));
+  });
+
+  testWidgets('can select all non-mac', (WidgetTester tester) async {
+    final FocusNode node = FocusNode();
+    await tester.pumpWidget(MaterialApp(
+      home: SelectionArea(
+        focusNode: node,
+        selectionControls: materialTextSelectionControls,
+        child: ListView.builder(
+          itemCount: 100,
+          itemBuilder: (BuildContext context, int index) {
+            return Text('Item $index');
+          },
+        ),
+      ),
+    ));
+    await tester.pumpAndSettle();
+    node.requestFocus();
+    await tester.sendKeyDownEvent(LogicalKeyboardKey.controlLeft);
+    await tester.sendKeyDownEvent(LogicalKeyboardKey.keyA);
+    await tester.sendKeyUpEvent(LogicalKeyboardKey.keyA);
+    await tester.sendKeyUpEvent(LogicalKeyboardKey.controlLeft);
+    await tester.pump();
+
+    for (int i = 0; i < 13; i += 1) {
+      final RenderParagraph paragraph = tester.renderObject<RenderParagraph>(find.descendant(of: find.text('Item $i'), matching: find.byType(RichText)));
+      expect(paragraph.selections[0], TextSelection(baseOffset: 0, extentOffset: 'Item $i'.length));
+    }
+    expect(find.text('Item 13'), findsNothing);
+  }, variant: const TargetPlatformVariant(<TargetPlatform>{ TargetPlatform.android, TargetPlatform.windows, TargetPlatform.linux, TargetPlatform.fuchsia }));
+
+  testWidgets('can select all mac', (WidgetTester tester) async {
+    final FocusNode node = FocusNode();
+    await tester.pumpWidget(MaterialApp(
+      home: SelectionArea(
+        focusNode: node,
+        selectionControls: materialTextSelectionControls,
+        child: ListView.builder(
+          itemCount: 100,
+          itemBuilder: (BuildContext context, int index) {
+            return Text('Item $index');
+          },
+        ),
+      ),
+    ));
+    await tester.pumpAndSettle();
+
+  }, variant: const TargetPlatformVariant(<TargetPlatform>{ TargetPlatform.iOS, TargetPlatform.macOS }));
```

Comment:
Is this test incomplete or am I misunderstanding?

Reply:
oh it looks like i missed it, nice catch!